src/plugins/intel_gpu/src/plugin/transformations/reduce_modification_for_false_keepdims.cpp

```diff
@@ -0,0 +1,115 @@
+    // Get one MatcherPass for all modes
+    auto reduce_pattern = ngraph::pattern::wrap_type<ngraph::opset10::ReduceSum,
+                                                     ngraph::opset10::ReduceMean,
+                                                     ngraph::opset10::ReduceProd,
+                                                     ngraph::opset10::ReduceMin,
+                                                     ngraph::opset10::ReduceMax>(
+        {ngraph::pattern::any_input(ngraph::pattern::has_static_shape()),
+         ngraph::pattern::wrap_type<ngraph::opset10::Constant>()},
+        ngraph::pattern::has_static_shape());
+
+    // register callback
+    ov::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        auto reduce =
+            as_type_ptr<op::util::ArithmeticReductionKeepDims>(pattern_map.at(reduce_pattern).get_node_shared_ptr());
+        if (!reduce)
+            return false;
+
+        auto input = reduce->input_value(0);
+        const auto input_shape = input.get_shape();
+        const auto reduce_shape = reduce->output(0).get_shape();
+        const auto input_rank = input.get_partial_shape().rank().get_length();
+
+        auto axes_vector = reduce->get_reduction_axes().to_vector();
+        std::sort(axes_vector.begin(), axes_vector.end());
+
+        if (!reduce->get_keep_dims() && is_unsupported_reordered_axes(axes_vector, input_rank, (input_rank - 2)) &&
+            input_shape.size() < 6) {
+            ngraph::NodeVector new_ops;
+
+            // Reduce
+            auto reduce_const =
+                ngraph::opset10::Constant::create(ngraph::element::i64, ngraph::Shape{axes_vector.size()}, axes_vector);
+
+            // Add each reduce mode supported by oneDNN
+            if (ngraph::is_type<ngraph::opset10::ReduceSum>(reduce))
+                input = std::make_shared<ngraph::opset10::ReduceSum>(input, reduce_const, true);
+            else if (ngraph::is_type<ngraph::opset10::ReduceMean>(reduce))
+                input = std::make_shared<ngraph::opset10::ReduceMean>(input, reduce_const, true);
+            else if (ngraph::is_type<ngraph::opset10::ReduceMin>(reduce))
+                input = std::make_shared<ngraph::opset10::ReduceMin>(input, reduce_const, true);
+            else if (ngraph::is_type<ngraph::opset10::ReduceMax>(reduce))
+                input = std::make_shared<ngraph::opset10::ReduceMax>(input, reduce_const, true);
+            else if (ngraph::is_type<ngraph::opset10::ReduceProd>(reduce))
+                input = std::make_shared<ngraph::opset10::ReduceProd>(input, reduce_const, true);
+
+            input.get_node_shared_ptr()->set_friendly_name(reduce->get_friendly_name());
+            new_ops.push_back(input.get_node_shared_ptr());
+
+            // Reshape
+            auto reshape_shape = ngraph::Shape(input_shape.size(), 1);
+            // Expected that a feature axis is only un-reduced.
```

Comment:
what about adding assertion here to assure that output is 1d?

Reply:
I mean, not `assert(input_shape.size() != 1);`, but `assert(reduce_shape.size() == 1);`